k9mail/src/main/java/security/planck/provisioning/ProvisioningManager.kt

```diff
@@ -64,6 +69,51 @@ class ProvisioningManager @Inject constructor(
         }
     }
 
+    fun initializeApp() {
+        startProvisioning()
+    }
+
+    suspend fun restoreData(documentFile: DocumentFile?) {
```

Comment:
was the race condition caused by the function not being suspend before?

Reply:
The problem is that I was calling the `viewModel.initializeApp()` from the main thread, which is not a bad thing since it calls the suspend funcrtion `ProvisioningManager.startProvisioning()`.
The actual problem seemed to be that I was calling the said `viewModel.initializeApp()` from the response of the user choosing the folder where to import dbs from.
Now the change is to call same function after all the files have been copied. Because since the copying files is also a suspend operation, they must have been having race conditions between both operations.